Content.Shared/Roles/SharedRoleSystem.cs

```diff
@@ -31,6 +33,15 @@ public override void Initialize()
         Subs.CVar(_cfg, CCVars.GameRoleTimerOverride, SetRequirementOverride, true);
 
         SubscribeLocalEvent<MindRoleComponent, ComponentShutdown>(OnComponentShutdown);
+        SubscribeLocalEvent<StartingMindRoleComponent, PlayerSpawnCompleteEvent>(OnSpawn);
+    }
+
+    private void OnSpawn(EntityUid uid, StartingMindRoleComponent component, PlayerSpawnCompleteEvent args)
+    {
+        if (!_minds.TryGetMind(uid, out var mindId, out _))
+            return;
+
+        MindAddRole(mindId, component.MindRole , silent:component.Silent);
```

Comment:
```suggestion
        if (!_minds.TryGetMind(uid, out var mindId, out var mindComp))
            return;

        MindAddRole(mindId, component.MindRole , mind: mindComp, silent: component.Silent);
```
We get the component anyways, so we can also pass it into the function here so we don't have to do it a second time.